### PR TITLE
seapath_basic_cluster_definition_example: basic template to make a cluster 

### PR DESCRIPTION
**Pull request by TanguyRaufflet**
Add inventory based on the example template. But simplified, to create a very basic cluster for POC.
Add documentation on the variables to modify on the template to use it and a schematic to describe the configuration of the hosts (use of a switch)

### CODE REVIEW

File: inventories/README.md [hunk 1]
```diff
@@ -10,6 +10,24 @@ We recommend to have 3 kinds of inventories:
 - The OpenVSwitch topology inventory, where you will describe the different bridges and ports configuration for OVS: a template is provided (seapath_ovstopology_definition_example.yml)
 - The VM inventory, where you will describe all variables related to the virtual machine (number of cores, features, ip, network interface, etc.): a template is provided (seapath_vm_definition_example.yml)
 
+## Basic cluster inventory
+
+A basic template of inventory to form a basic cluster for POC is provided (seapath_basic_cluster_definition_example.yml)
+
+<img src="./basic_cluster.png" alt="Exemple d'image" style="max-width:500px">
+
```
Comment (dupremathieu) [inline inventories/README.md:18]: Add a warning to indicate that this basic cluster does not support:
* low latency network
* dedicated cluster network with redundancy
* all advanced features like:
  * SNMP
  * syslog log sending 
  * advanced isolation

**Review by dupremathieu — CHANGES_REQUESTED** [review]

File: inventories/README.md (hunk 1 above)
Comment (TanguyRaufflet) [reply inventories/README.md:27]: Done

File: inventories/README.md [hunk 2]
```diff
@@ -2,30 +2,39 @@
...
-- The basic cluster inventory for POC: seapath_basic_cluster_definition_example.yml
-- A standalone description inventory: seapath_standalone_definition_example
-- The OpenVSwitch topology inventory, where you will describe the different bridges and ports configuration for OVS: seapath_ovstopology_definition_example.yml
-- The VM inventory, where you will describe all variables related to the virtual machine (number of cores, features, ip, network interface, etc.): seapath_vm_definition_example.yml
-- The firewall rules, where you will describe the packet filtering and the NAT on the system: iptables_rules_example.txt
+On SEAPATH, the definition of our setup must be described inside [Ansible inventories](https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html#passing-multiple-inventory-sources).
+You have to define your own inventories files which match with your setup.
+
+In this directory, you can find some commented examples of inventories for the following case:
+* a demo SEAPATH cluster which can be used to easily test SEAPATH with retails machines (use it as a POC, not in production)
```
Comment (Paullgk) [inline inventories/README.md:11]: A Seapath cluster demo

File: inventories/README.md [hunk 3]
```diff
@@ -2,30 +2,39 @@
...
+* a complex SEAPATH cluster
+
+It is possible with Ansible to provides [multiple inventories files](https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html#passing-multiple-inventory-sources), and we recommend using this feature for complex setup. In this directory, we have separated the "complex SEAPATH cluster" into three inventories. We have a dedicated inventory file for:
+* the physical machines
+* the cluster internal network (Open vSwitch topology)
+* the virtual machines
+## SEAPATH Ansible inventories examples description
+
+All inventories examples are in YAML format. The example inventories are:
+* `seapath_demo_cluster_definition_example.yml`: demo SEAPATH cluster
```
Comment (Paullgk) [inline inventories/README.md:22]: SEAPATH cluster demo

File: inventories/README.md [hunk 4]
```diff
@@ -2,30 +2,39 @@
...
+* `seapath_standalone_definition_example`: SEAPATH standalone hypervisor setup
+* The complex SEAPATH cluster inventories are:
+	* `seapath_cluster_definition_example.yml`: physical machines
+	* `seapath_ovstopology_definition_example.yml`: cluster internal network (Open vSwitch topology)
+	* `seapath_vm_definition_example.yml`: virtual machines
 
+The directory also contains the firewall rules `iptables_rules_example.txt` which are referring in `seapath_cluster_definition_example.yml`.
 
-## Optionnal variables
+Except the simple SEAPATH cluster inventory which require minor tweak to be adapted, in all other files we described all the options which do not match a realistic case.
```
Comment (Paullgk) [inline inventories/README.md:31]: minors tweaks

>  in all other files we described all the options which do not match a realistic case.

Strange formulation difficult to understand

File: inventories/README.md [hunk 5]
```diff
@@ -40,13 +49,15 @@ remove_all_networkd_config: true
 # THIS MUST NOT BE USED WITH skip_recreate_team0_config at the same time or the cluster network config won't be recreated.
 ```
 
-## Basic cluster inventory
+### Demo SEAPATH cluster inventory
+
+If we are SEAPATH newcomer, we can use this example to create a SEAPATH POC cluster. At the opposite of a real SEAPATH cluster, which requires an advance machine with multiple cores and network cards. This example only requires one network interface and can be made with costumer retail machines. It had to be tested by the SEAPATH team with Intel NUC machines.
```
Comment (Paullgk) [inline inventories/README.md:54]: If you are a

Comment (Paullgk) [inline inventories/README.md:54]: > It had to be tested by the SEAPATH team with Intel NUC machines.

I don't think this sentence is needed here

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/README.md [hunk 6]
```diff
@@ -2,17 +2,49 @@
 <!-- Copyright (C) 2024, SFL (https://savoirfairelinux.com) -->
 <!-- SPDX-License-Identifier: CC-BY-4.0 -->
 
+# Ansible configuration
+Configuring the cluster through ansible requires inventories and we recommand 3 kinds:
+- The description of the machines that will form the cluster
+- The description of the OpenVSwitch network
+- The description of the virtual machines
+In addition to this a file to configure the firewall rules for the cluster's network is needed
```
Comment (Paullgk) [inline inventories/README.md]: missing final point

File: inventories/README.md [hunk 7]
```diff
@@ -2,17 +2,49 @@
...
+
 # Inventories directory
 
 This is the place where you can store your inventories.
-We recommend to have 3 kinds of inventories:
-- The cluster description inventory, where you will populate the details of your nodes (ip, interface, network details, disks, etc.): a template is provided (seapath_cluster_definition_example.yml)
-- The OpenVSwitch topology inventory, where you will describe the different bridges and ports configuration for OVS: a template is provided (seapath_ovstopology_definition_example.yml)
-- The VM inventory, where you will describe all variables related to the virtual machine (number of cores, features, ip, network interface, etc.): a template is provided (seapath_vm_definition_example.yml)
+In this folder, different templates are provided:
+- The cluster description inventory, where you will populate the details of your nodes (ip, interface, network details, disks, etc.): seapath_cluster_definition_example.yml
```
Comment (Paullgk) [inline inventories/README.md]: I recommend to put each file name in a code beacon: `seapath_cluster_definition_example.yml`

File: inventories/README.md [hunk 8]
```diff
@@ -2,17 +2,49 @@
...
+- The basic cluster inventory for POC: seapath_basic_cluster_definition_example.yml
+- A standalone description inventory: seapath_standalone_definition_example
```
Comment (Paullgk) [inline inventories/README.md]: Missing .yaml extension

File: inventories/README.md [hunk 9]
```diff
@@ -2,17 +2,49 @@
...
+- The OpenVSwitch topology inventory, where you will describe the different bridges and ports configuration for OVS: seapath_ovstopology_definition_example.yml
+- The VM inventory, where you will describe all variables related to the virtual machine (number of cores, features, ip, network interface, etc.): seapath_vm_definition_example.yml
+- The firewall rules, where you will describe the packet filtering and the NAT on the system: iptables_rules_example.txt
+
+
+## Optionnal variables
+
+You can find concrete examples of the variables in the inventories of this directory. Below are optionnal advanced variables that are not described in the examples.
```
Comment (Paullgk) [inline inventories/README.md]: optional

**Review by Paullgk — CHANGES_REQUESTED** [review]

**Review by Paullgk — CHANGES_REQUESTED** [review]
A good practice in git commit is 1 commit = 1 modification (better for reading latter what your commit does)
So I would suggest you to split this commit into 2 commits.

File: inventories/seapath_basic_cluster_definition_example.yml [hunk 10]
```diff
@@ -5,15 +5,16 @@ all:
             children:
                 hypervisors:
                     hosts:
```
Comment (Paullgk) [inline inventories/seapath_basic_cluster_definition_example.yml]: This review is concerning the commit message and the commit itself:

1. A good practice in git commit is 1 commit = 1 modification (better for reading latter what your commit does)
So I would suggest you to split this commit into 2 commits
2. On your commit title, a good practice is "<FILE NAME>: <COMMIT PURPOSE> (=what this commit does)"

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/README.md [hunk 11]
```diff
@@ -10,6 +10,33 @@ We recommend to have 3 kinds of inventories:
 - The OpenVSwitch topology inventory, where you will describe the different bridges and ports configuration for OVS: a template is provided (seapath_ovstopology_definition_example.yml)
```
Comment (Paullgk) [inline inventories/README.md]: In your commit message, missing final point in the commit msg

File: inventories/seapath_basic_cluster_definition_example.yml [hunk 12]
```diff
@@ -0,0 +1,151 @@
+---
```
Comment (Paullgk) [inline inventories/seapath_basic_cluster_definition_example.yml]: In your commit message, missing final point in the commit msg

File: inventories/README.md [hunk 13]
```diff
@@ -10,6 +10,33 @@ We recommend to have 3 kinds of inventories:
...
 - The VM inventory, where you will describe all variables related to the virtual machine (number of cores, features, ip, network interface, etc.): a template is provided (seapath_vm_definition_example.yml)
 
+## Basic cluster inventory
+
+A basic template of inventory to form a basic cluster for POC is provided (seapath_basic_cluster_definition_example.yml)
```
Comment (Paullgk) [inline inventories/README.md]: Missing final point + use code beacon for filename

File: inventories/README.md [hunk 14]
```diff
@@ -10,6 +10,33 @@ We recommend to have 3 kinds of inventories:
...
+
+#### ⚠ Do not support ⚠
```
Comment (Paullgk) [inline inventories/README.md]: What is the purpose of this section? Some explanation are needed

**Review by Paullgk — CHANGES_REQUESTED** [review]

**Review by Paullgk — CHANGES_REQUESTED** [review]
In your commit message: exemple --> example

File: inventories/seapath_basic_cluster_definition_example.yml (hunk 12 above)
Comment (Paullgk) [inline inventories/seapath_basic_cluster_definition_example.yml]: In your commit message: exemple --> example

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/README.md (hunk 5 above)
Comment (TanguyRaufflet) [reply inventories/README.md:54]: @dupremathieu what do you think of this ?

File: inventories/README.md (hunk 2 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Done

File: inventories/README.md (hunk 4 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: corrected by "The simple SEAPATH cluster inventory require only minor tweak to be adapted. In all other files described, all the options which do not match a realistic case and need to be modified are indicated."

File: inventories/README.md (hunk 5 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Done

Comment (TanguyRaufflet) [reply inventories/README.md]: sentence removed

File: inventories/README.md (hunk 7 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Corrected by last commit

File: inventories/README.md (hunk 8 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Done

File: inventories/README.md (hunk 9 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Corrected by last commit

File: inventories/seapath_basic_cluster_definition_example.yml [hunk 15]
```diff
@@ -20,15 +20,6 @@ all:
 
                     # hypervisors common vars
                     vars:
```
[the remark this answers is not in the document]
Comment (TanguyRaufflet) [reply inventories/seapath_basic_cluster_definition_example.yml]: Done

File: inventories/README.md (hunk 11 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Done

File: inventories/README.md (hunk 13 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Done + beacon corrected by last commit

File: inventories/seapath_basic_cluster_definition_example.yml (hunk 12 above)
Comment (TanguyRaufflet) [reply inventories/seapath_basic_cluster_definition_example.yml]: Done

File: inventories/README.md (hunk 6 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Done

File: inventories/README.md (hunk 14 above)
Comment (TanguyRaufflet) [reply inventories/README.md]: Corrected by last commit

File: inventories/seapath_basic_cluster_definition_example.yml (hunk 12 above)
Comment (TanguyRaufflet) [reply inventories/seapath_basic_cluster_definition_example.yml]: Done

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/README.md [hunk 16]
```diff
@@ -10,6 +10,33 @@ We recommend to have 3 kinds of inventories:
 - The OpenVSwitch topology inventory, where you will describe the different bridges and ports configuration for OVS: a template is provided (seapath_ovstopology_definition_example.yml)
 - The VM inventory, where you will describe all variables related to the virtual machine (number of cores, features, ip, network interface, etc.): a template is provided (seapath_vm_definition_example.yml)
```
Comment (Paullgk) [inline inventories/README.md]: In your commit msg: missing verb, should be "add a basic template to make a cluster"

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/seapath_basic_cluster_definition_example.yml [hunk 17]
```diff
@@ -5,15 +5,16 @@ all:
             children:
                 hypervisors:
```
Comment (Paullgk) [inline inventories/seapath_basic_cluster_definition_example.yml]: Your commit title is to general to understand quickly what it does. 
A good practice is <FILE NAME>: <COMMIT PURPOSE> (refactor documentation, etc)

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/README.md (hunk 6 above)
Comment (Paullgk) [reply inventories/README.md]: I still don't see the final point

File: inventories/README.md (hunk 7 above)
Comment (Paullgk) [reply inventories/README.md]: I don't see it on my side

File: inventories/README.md (hunk 8 above)
Comment (Paullgk) [reply inventories/README.md]: I don't see it on my side

File: inventories/README.md (hunk 9 above)
Comment (Paullgk) [reply inventories/README.md]: Not on my side

**Review by Paullgk — CHANGES_REQUESTED** [review]

File: inventories/README.md [hunk 18]
```diff
@@ -2,17 +2,49 @@
 <!-- Copyright (C) 2024, SFL (https://savoirfairelinux.com) -->
 <!-- SPDX-License-Identifier: CC-BY-4.0 -->
 
+# Ansible configuration
+Configuring the cluster through ansible requires inventories and we recommand 3 kinds:
+- The description of the machines that will form the cluster
```
Comment (Paullgk) [inline inventories/README.md]: Missing verb in your commit title

**Review by Paullgk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**